integration-cli/docker_cli_run_test.go

```diff
@@ -4507,3 +4507,53 @@ func (s *DockerDaemonSuite) TestRunWithUlimitAndDaemonDefault(c *check.C) {
 	c.Assert(err, checker.IsNil)
 	c.Assert(out, checker.Contains, "[nofile=42:42]")
 }
+
+func (s *DockerSuite) TestRunMountBind(c *check.C) {
+	testRequires(c, DaemonIsLinux, SameHostDaemon, NotUserNamespace)
+
+	// Prepare a source directory to bind mount
+	tmpDir, err := ioutil.TempDir("", "mount-bind")
+	if err != nil {
+		c.Fatal(err)
+	}
+	defer os.RemoveAll(tmpDir)
+
+	mnt1, mnt2 := path.Join(tmpDir, "mnt1"), path.Join(tmpDir, "mnt2")
+	if err := os.Mkdir(mnt1, 0755); err != nil {
+		c.Fatal(err)
+	}
+	if err := os.Mkdir(mnt2, 0755); err != nil {
+		c.Fatal(err)
+	}
+	if err := ioutil.WriteFile(path.Join(mnt1, "test1"), []byte("test1"), 0644); err != nil {
+		c.Fatal(err)
+	}
+	if err := ioutil.WriteFile(path.Join(mnt2, "test2"), []byte("test2"), 0644); err != nil {
+		c.Fatal(err)
+	}
+
+	cName := "mount-bind"
+	dockerCmd(c, "run", "-i", "-d", "--name", cName,
+		"--mount", fmt.Sprintf("type=bind,src=%s,dst=/foo", mnt1),
+		"--mount", fmt.Sprintf("type=bind,src=%s,dst=/bar", mnt2),
+		"busybox", "top")
+
+	out, _ := dockerCmd(c, "exec", cName, "cat", "/foo/test1")
+	if out != "test1" {
+		c.Fatalf("mnt1 (foo) not mounted")
+	}
+	out, _ = dockerCmd(c, "exec", cName, "cat", "/bar/test2")
+	if out != "test2" {
+		c.Fatalf("mnt2 (bar) not mounted")
+	}
+}
+
+func (s *DockerSuite) TestRunMountBindBad(c *check.C) {
+	testRequires(c, DaemonIsLinux, SameHostDaemon, NotUserNamespace)
+	out, exitCode, _ := dockerCmdWithError("run",
+		"-v", "/foo:/foo1",
+		"--mount", "type=bind,src=/bar,dst=/bar1",
+		"busybox", "top")
+	c.Assert(exitCode, checker.Not(checker.Equals), 0)
+	c.Assert(out, checker.Contains, "must not specify both Binds and Mounts")
```

Comment:
The error was defined in https://github.com/docker/docker/pull/22373/files#diff-2660508eef7a8eeadb6d27b8bbf6d718R78 rather than in this PR.
IMO the error message is correct because `-v /foo:/foo1` actually sets `HostConfig.Binds` rather than `Config.Volumes`.
But I agree we should improve the message.
CC @cpuguy83